src/coprocessor/dag/rpn_expr/impl_arithmetic.rs

```diff
@@ -313,6 +314,95 @@ impl ArithmeticOp for DecimalMultiply {
     }
 }
 
+#[derive(Debug)]
+pub struct IntDivideInt;
+
+impl ArithmeticOp for IntDivideInt {
+    type T = Int;
+
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        if *rhs == 0 {
+            return Ok(None);
+        }
+        Ok(Some(div_i64(*lhs, *rhs)?))
+    }
+}
+
+#[derive(Debug)]
+pub struct IntDivideUint;
+
+impl ArithmeticOp for IntDivideUint {
+    type T = Int;
+
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        if *rhs == 0 {
+            return Ok(None);
+        }
+        Ok(Some(div_i64_with_u64(*lhs, *rhs as u64).map(|r| r as i64)?))
+    }
+}
+
+#[derive(Debug)]
+pub struct UintDivideUint;
+
+impl ArithmeticOp for UintDivideUint {
+    type T = Int;
+
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        if *rhs == 0 {
+            return Ok(None);
+        }
+        Ok(Some(((*lhs as u64) / (*rhs as u64)) as i64))
+    }
+}
+
+#[derive(Debug)]
+pub struct UintDivideInt;
+
+impl ArithmeticOp for UintDivideInt {
+    type T = Int;
+
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        if *rhs == 0 {
+            return Ok(None);
+        }
+        Ok(Some(div_u64_with_i64(*lhs as u64, *rhs).map(|r| r as i64)?))
+    }
+}
+
+#[rpn_fn(capture = [ctx])]
+#[inline]
+fn int_divide_decimal(
+    ctx: &mut EvalContext,
+    lhs: &Option<Decimal>,
+    rhs: &Option<Decimal>,
+) -> Result<Option<Decimal>> {
```

Comment:
Looks like the original `int_divide_decimal` implementation returns `i64` instead of `Decimal`.